src/System/Process/Typed.hs

```diff
@@ -1142,3 +1145,39 @@ instance Exception ByteStringOutputException
 -- @since 0.1.1
 unsafeProcessHandle :: Process stdin stdout stderr -> P.ProcessHandle
 unsafeProcessHandle = pHandle
+
+-- Ported from unliftio
+
+bracket :: MonadUnliftIO m => m a -> (a -> m b) -> (a -> m c) -> m c
```

Comment:
I hadn't thought through this part of it. There are subtle differences between the `bracket` and `finally` in `unliftio` and `base` with regards to masking state. I'd pictured using `Control.Exception.bracket` and throwing a `withRunInIO` around it, but this works too. I'm OK with leaving this code as-is, but if you're interested in trying the other way, that's also a possibility, and will result in shorter code.